### ISSUE TITLE
klist fail when running via jamf

### ISSUE THREAD

**Issue by franton** [pbowden-msft/SignInHelper#1]
Hi Paul!

So tregerl on macadmin slack found this. When you run your script from Jamf via a policy, it runs in a non user context thus things like klist commands fail to work as expected. The cure is very simple thankfully.

After L41 where you detect the user, insert the following:
`local LOGGEDINID=$(id -u $LOGGEDIN)`

Now change your various klist lines so they have `launchctl runas $LOGGEDIN`
e.g.
`local KERB=$(${CMD_PREFIX} launchctl asuser $LOGGINID /usr/bin/klist 2> /dev/null)`

**Comment by georgeescott** [comment]
I also ran in to the same issue with Jamf. 
```
Script result: >>INFO - Script is running in the root security context - running commands as user: usernamehere
>>INFO - Detected that this machine is domain joined
>>WARNING - No Kerberos cache present
>>ERROR - Could not detect UPN
```

If I tried to run `klist` via SSH on the Mac while the user was logged in, it would return `klist: Cache not found`.

Turns out that you can't run `klist` via SSH/script or as root in newer macOS versions: [https://macadmins.slack.com/archives/C04QVP86E/p1526641453000228](https://macadmins.slack.com/archives/C04QVP86E/p1526641453000228), it has to be in an active GUI session, or run as the user.

On [L372](https://github.com/pbowden-msft/SignInHelper/blob/2d11433fdd0148f545237620a8a8473a55ff4b00/SignInHelper#L372) `CMD_PREFIX="/usr/bin/sudo -u ${CURRENTUSER}"`, it seems `sudo -u` no longer works for Kerberos commands on newer macOS versions.

Reading [https://scriptingosx.com/2020/08/running-a-command-as-another-user/](https://scriptingosx.com/2020/08/running-a-command-as-another-user/), we can add `launchctl asuser` as you suggest.

You were close for your changes but you only need to change `CMD_PREFIX`, rather than adding to every `klist` line.

Make the following changes to fix this issue:
1. Before [L372](https://github.com/pbowden-msft/SignInHelper/blob/2d11433fdd0148f545237620a8a8473a55ff4b00/SignInHelper#L372), add:
`CURRENTUSERID=$(id -u "$CURRENTUSER")`
2. Then on what will be L373, change `CMD_PREFIX="/usr/bin/sudo -u ${CURRENTUSER}"` to:
`CMD_PREFIX="/bin/launchctl asuser ${CURRENTUSERID} /usr/bin/sudo -u ${CURRENTUSER}"`

**Comment by pbowden-msft** [comment]
@georgeescott Thanks! I merged your changes into v1.6.0 of the tool.
